### PR TITLE
feat: getBrandInfo using ChainHub

### LINKED ISSUE #9211: bidirectional lookup of denoms / brands for IBC
## What is the Problem Being Solved?

https://github.com/Agoric/agoric-sdk/issues/8879 manages IBC paths but developers need a way to establish a correspondence between a denom and its issuing chain, and optionally an ERTP Brand.

## Description of the Design

- Treat channel info as immutable
- We provide a lookup service to get the info for a particular chain
- The set is append-only (the value for a particular chain name is immutable)
- The right to append is held only by BLD Stakers
- If a developer needs the info before it’s on-chain (between staker votes) they look it up through some other trusted source (e.g. Google for the chain’s docs)

With the above requirements we landed on each contract having a ChainHub. That can load from agoricNames and/or be filled manually by contract code.

## Test Plan

`getBrandInfo` works with ChainHub (done in #9754)

All exos look up denoms using ChainHub

Not in scope: filling ChainHub using agoricNames, which is
-  #9752

## Upgrade Considerations
not yet deployed but needs an upgrade test

### PR DESCRIPTION
**Pull request by dckc**
closes #9211

## Description

  - feat: chainHub.registerAsset / lookupAsset
  - feat: Orchestrator.getBrandInfo
    - getChain() is now backed by a store
  - groundwork for asset info in agoricNames  - #9752

### Testing Considerations

unit tests cover happy path.

Low code-complexity doesn't seem to merit more.

### Security / Scaling Considerations

none, AFAICT

### Documentation Considerations

make `getBrandInfo()` work as documented, provided assets are registered with the ChainHub

### Upgrade Considerations

not yet deployed

### CODE REVIEW

File: packages/orchestration/src/exos/chain-hub.js
Comment (turadg) [inline packages/orchestration/src/exos/chain-hub.js:34]: this works it's a bit circular. since this type is used only for the guard, shall we keep it with the guard? That would suggest moving this to the `typeGuards.js` module, but I notice this shape is used only in this exo so we shouldn't export it. Instead please move the guard here.

Another reason that's better than moving the type out is so the properties descriptions are easy to access. Looking it more I see that TS LSP isn't picking up the the jsdoc. 
Screenshot 2024-07-25 at 11 17 13 AM

I tried changing to `.ts syntax` and that doesn't solve it.

File: packages/orchestration/src/exos/chain-hub.js
Comment (turadg) [inline packages/orchestration/src/exos/chain-hub.js]: for consistency,
```suggestion
  const denomDetails = zone.mapStore('denom', {
```

File: packages/orchestration/src/exos/chain-hub.js
Comment (turadg) [inline packages/orchestration/src/exos/chain-hub.js]: Confused by this special case. Doesn't `chainInfos` have 'agoric'?

File: packages/orchestration/src/exos/orchestrator.js
Comment (turadg) [inline packages/orchestration/src/exos/orchestrator.js]: I think we import from `@endo/errors` now

File: packages/orchestration/src/exos/orchestrator.js
Comment (turadg) [inline packages/orchestration/src/exos/orchestrator.js]: I don't know but please address before merge

File: packages/orchestration/src/exos/orchestrator.js
Comment (turadg) [inline packages/orchestration/src/exos/orchestrator.js:150]: this is surprising. `getFoo` sounds idempotent. Is there away around this? If not, shall we rename `getChain`?

File: packages/orchestration/test/facade.test.ts
Comment (turadg) [inline packages/orchestration/test/facade.test.ts:136]: what failed without this? I wonder if we're better off addressing it. I could resume ,
- https://github.com/Agoric/agoric-sdk/pull/9677

File: packages/orchestration/src/exos/chain-hub.js
Comment (dckc) [reply packages/orchestration/src/exos/chain-hub.js:34]: ok; done

File: packages/orchestration/src/exos/chain-hub.js
Comment (dckc) [reply packages/orchestration/src/exos/chain-hub.js]: ok, I suppose it should

File: packages/orchestration/src/exos/orchestrator.js
Comment (dckc) [reply packages/orchestration/src/exos/orchestrator.js:150]: It's part of the "you have to tell it all the answers before you ask the question" design that lets `getBrandInfo()` be sync.

I could go with renaming `getChain` -> `provideChain`, provided the keepers of `orchestration-api.d.ts` agree.

File: packages/orchestration/test/facade.test.ts
Comment (dckc) [reply packages/orchestration/test/facade.test.ts:136]: I was just following the pattern of other tests in this file.

If I take it out, I get

```
  test/facade.test.ts:137

   136:                                           
   137:   const { zcf } = await setupZCFTest();   
   138:   const zone = provideDurableZone('test');

  Rejected promise returned by test. Reason:

  Error {
    message: 'value is not durable: {"bundleCap":"[Alleged: FakeBundleCapHandle]","bundleID":"b1-contract"} at slot 0 of "#{\\"bundleCap\\":\\"$0.Alleged: FakeBundleCapHandle\\",\\"bundleID\\":\\"b1-contract\\"}"',
  }
```

Comment (turadg) [reply packages/orchestration/test/facade.test.ts:136]: fair. When 9677 is ready I'll just go look for reincarnations that can be removed

File: packages/orchestration/src/exos/orchestrator.js
Comment (dckc) [reply packages/orchestration/src/exos/orchestrator.js]: seems to be not needed

File: packages/orchestration/src/exos/orchestrator.js
Comment (turadg) [reply packages/orchestration/src/exos/orchestrator.js:150]: Ok, out of scope

File: packages/orchestration/src/chain-info.js
Comment (turadg) [inline packages/orchestration/src/chain-info.js]: please source from a constant,
https://github.com/Agoric/agoric-sdk/blob/3845eacd2691a7dfadd0f96aa723827ae07153e9/packages/orchestration/src/exos/chain-hub.js#L25-L29

The other `provideChild` calls in this function should have been as well. Please improve while here

File: packages/orchestration/src/exos/chain-hub.js
Comment (turadg) [inline packages/orchestration/src/exos/chain-hub.js]: this looks to be imported
```suggestion
     * @param {Denom} denom - on the holding
```

File: packages/orchestration/src/exos/orchestrator.js
Comment (turadg) [inline packages/orchestration/src/exos/orchestrator.js]: extra whitespace

File: packages/orchestration/test/exos/chain-hub.test.ts
Comment (turadg) [inline packages/orchestration/test/exos/chain-hub.test.ts]: consider mentioning https://github.com/Agoric/agoric-sdk/issues/9752

**Review by turadg — APPROVED** [review]
LGTM contingent on the "please" comments

File: packages/orchestration/test/exos/chain-hub.test.ts
Comment (0xpatrickdev) [inline packages/orchestration/test/exos/chain-hub.test.ts:148]: Is this supposed to be agoric(cosmoshub(kava-usdt))?

File: packages/orchestration/src/cosmos-api.ts
Comment (0xpatrickdev) [inline packages/orchestration/src/cosmos-api.ts:77]: Do we plan to put all of this in vstorage? Seems we're currently only using `base` and `traces`?

File: packages/orchestration/src/exos/chain-hub.js
Comment (0xpatrickdev) [inline packages/orchestration/src/exos/chain-hub.js:411]: Is the plan to register a representation of an asset on every chain (that it's connected to)? And we'll only end up using `denomHash()` in tests / scripts to build proposals?

**Review by 0xpatrickdev — APPROVED** [review]
LGTM. Left some questions to better understand the design being worked towards

File: packages/orchestration/test/exos/chain-hub.test.ts
Comment (dckc) [reply packages/orchestration/test/exos/chain-hub.test.ts:148]: no, just cosmoshub(kava-usdt)

https://github.com/cosmos/chain-registry/blob/master/cosmoshub/assetlist.json#L42

File: packages/orchestration/src/cosmos-api.ts
Comment (dckc) [reply packages/orchestration/src/cosmos-api.ts:77]: Copying all of [cosmoshub/assetlist.json](https://github.com/cosmos/chain-registry/blob/master/cosmoshub/assetlist.json) is easier than thinking.

Seems like lots of it is useful for UIs and such. I think I heard @dtribble  say other chains were talking about putting more of this metadata on-chain.

But I don't feel strongly one way or another. It's only a test fixture, so far.

File: packages/orchestration/src/exos/chain-hub.js
Comment (dckc) [reply packages/orchestration/src/exos/chain-hub.js:411]: Yes, that seems like the simplest thing that can work, based on discussion with @turadg earlier this week.

Maybe not every asset, but every asset they put in the chain registry.

**Pull request merged**
